opa/opa.go

```diff
@@ -0,0 +1,201 @@
+/*
+import (
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo-contrib/opaecho"
+)
+
+func main() {
+	e := echo.New()
+
+	module := `
+		package example.authz
+		default allow := false
+		allow {
+			input.Method == "GET"
+		}
+	`
+
+	cfg := opaecho.Config{
+		RegoQuery:             "data.example.authz.allow",
+		RegoPolicy:            bytes.NewBufferString(module),
+		IncludeQueryString:    true,
+		DeniedStatusCode:      http.StatusForbidden,
+		DeniedResponseMessage: "status forbidden",
+		IncludeHeaders:        []string{"Authorization"},
+	}
+
+	e.Use(opaecho.MiddlewareWithConfig(cfg))
+
+	e.GET("/", func(c echo.Context) error {
+		return c.String(http.StatusOK, "Hello, World!")
+	})
+
+	e.Logger.Fatal(e.Start(":8080"))
+}
+*/
+
+
+
+
+package opaecho
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	"github.com/open-policy-agent/opa/rego"
+)
+
+type (
```

Comment:
please do not use `type ()` and `var ()` blocks - these are really annoying when it comes to copying some parts of code  and add unnecessary indentation depth